kt/plugins/godot-gradle-plugin/src/main/kotlin/godot/gradle/setupConfigurationsAndCompilations.kt

```diff
@@ -37,125 +40,194 @@ fun Project.setupConfigurationsAndCompilations(godotExtension: GodotExtension, j
         }
     }
 
-    with(tasks) {
-        val createBuildLock = with(create("createBuildLock")) {
-            doFirst {
-                val buildLockDir = getBuildLockDir(projectDir)
-                File(buildLockDir, "buildLock.lock").createNewFile()
+    afterEvaluate {
+        with(tasks) {
+            val createBuildLock = with(create("createBuildLock")) {
+                doFirst {
+                    val buildLockDir = getBuildLockDir(projectDir)
+                    File(buildLockDir, "buildLock.lock").createNewFile()
+                }
             }
-        }
 
-        val deleteBuildLock = with(create("deleteBuildLock")) {
-            doLast {
-                val buildLockDir = getBuildLockDir(projectDir)
-                File(buildLockDir, "buildLock.lock").delete()
+            val deleteBuildLock = with(create("deleteBuildLock")) {
+                doLast {
+                    val buildLockDir = getBuildLockDir(projectDir)
+                    File(buildLockDir, "buildLock.lock").delete()
+                }
             }
-        }
 
-        val bootstrapJar = with(create("bootstrapJar", ShadowJar::class.java)) {
-            archiveBaseName.set("godot-bootstrap")
-            exclude("**/module-info.class") //for android support: excludes java 9+ module info which cannot be parsed by the dx tool
-            configurations.clear()
-            configurations.add(bootstrapConfiguration)
+            val bootstrapJar = with(create("bootstrapJar", ShadowJar::class.java)) {
+                archiveBaseName.set("godot-bootstrap")
+                configurations.clear()
+                configurations.add(bootstrapConfiguration)
 
-            dependsOn(createBuildLock)
-        }
+                dependsOn(createBuildLock)
+            }
 
-        val shadowJar = with(named("shadowJar", ShadowJar::class.java).get()) {
-            archiveBaseName.set("main")
-            archiveVersion.set("")
-            archiveClassifier.set("")
-            exclude("**/module-info.class") //for android support: excludes java 9+ module info which cannot be parsed by the dx tool
+            val shadowJar = with(named("shadowJar", ShadowJar::class.java).get()) {
+                archiveBaseName.set("main")
+                archiveVersion.set("")
+                archiveClassifier.set("")
 
-            dependencies {
-                it.exclude(it.dependency("org.jetbrains.kotlin:kotlin-stdlib.*"))
-                it.exclude(it.dependency("com.utopia-rise:godot-library:.*"))
-            }
+                dependencies {
+                    it.exclude(it.dependency("org.jetbrains.kotlin:kotlin-stdlib.*"))
+                    it.exclude(it.dependency("com.utopia-rise:godot-library:.*"))
```

Comment:
Do we really need to remove our library here given it's compileOnly in the first place ?

Reply:
Yes because theoretically one can add our library as well in the `dependencies` block manually to whatever he wants. For example one can set it as `api` so it's a transitive dependency. Which is completely valid. But even then we don't want it to be included in the final jar.